File: jscomp/test/boolean_coercion_test.js
'use strict';

var Mt    = require("./mt");
var Block = require("../../lib/js/block");

var suites = [/* [] */0];

var test_id = [0];

function eq(loc, x, y) {
  test_id[0] = test_id[0] + 1 | 0;
  suites[0] = /* :: */[
    /* tuple */[
      loc + (" id " + test_id[0]),
      function () {
        return /* Eq */Block.__(0, [
                  x,
                  y
                ]);
      }
    ],
    suites[0]
  ];
  return /* () */0;
}

eq('File "boolean_coercion_test.ml", line 11, characters 7-14', /* true */1, ( !x ? 0 : 1 ));

eq('File "boolean_coercion_test.ml", line 12, characters 7-14', true, ( !! x));

eq('File "boolean_coercion_test.ml", line 13, characters 7-14', false, ( !! x));

eq('File "boolean_coercion_test.ml", line 14, characters 7-14', /* false */0, ( !x ? 0 : 1 ));

Mt.from_pair_suites("boolean_coercion_test.ml", suites[0]);

exports.suites  = suites;
exports.test_id = test_id;
exports.eq      = eq;
/*  Not a pure module */

File: jscomp/test/boolean_coercion_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = 
  incr test_id ; 
  suites := 
    (loc ^" id " ^ (string_of_int !test_id), (fun _ -> Mt.Eq(x,y))) :: !suites



let ()  =
    eq __LOC__ true (Js.Boolean.unsafe_bool_of_any 3);
    eq __LOC__ Js.true_ (Js.Boolean.unsafe_boolean_of_any 3);
    eq __LOC__ Js.false_ (Js.Boolean.unsafe_boolean_of_any 0);
    eq __LOC__ false (Js.Boolean.unsafe_bool_of_any 0)


let () = 
  Mt.from_pair_suites __FILE__ !suites      



